packages/web-components/src/radio/radio.styles.ts

```diff
@@ -0,0 +1,120 @@
+  colorNeutralStrokeAccessiblePressed,
+  colorStrokeFocus1,
+  colorStrokeFocus2,
+  fontFamilyBase,
+  fontSizeBase300,
+  fontWeightRegular,
+  lineHeightBase300,
+  spacingHorizontalS,
+  spacingHorizontalXS,
+  spacingVerticalS,
+} from '../theme/design-tokens.js';
+
+/** Radio styles
+ * @public
+ */
+export const styles = css`
+  ${display('inline-grid')}
+
+  :host {
+    grid-auto-flow: column;
+    grid-template-columns: max-content;
+    gap: ${spacingHorizontalXS};
+    align-items: center;
+    height: 32px;
+    cursor: pointer;
+    outline: none;
+    position: relative;
+    user-select: none;
+    color: var(--state-color);
+  }
+  :host([disabled]) {
+    color: ${colorNeutralForegroundDisabled};
+    pointer-events: none;
+  }
+  .label {
+    cursor: pointer;
+    font-family: ${fontFamilyBase};
+    font-size: ${fontSizeBase300};
+    font-weight: ${fontWeightRegular};
+    line-height: ${lineHeightBase300};
+  }
+  .label__hidden {
+    display: none;
+  }
+  .control {
+    box-sizing: border-box;
+    align-items: center;
+    border: 1px solid var(--control-border-color);
+    border-radius: ${borderRadiusCircular};
+    display: flex;
+    height: 16px;
+    justify-content: center;
+    margin: ${spacingVerticalS} ${spacingHorizontalS};
+    position: relative;
+    width: 16px;
+    justify-self: center;
+  }
+  .checked-indicator {
+    border-radius: ${borderRadiusCircular};
+    height: 10px;
+    opacity: 0;
+    width: 10px;
+  }
+  :host([aria-checked='false']:hover:not([disabled])) .control {
+    color: ${colorNeutralForeground2};
+  }
+  :host(:focus-visible)::after {
+    border: 2px solid ${colorStrokeFocus1};
+    border-radius: ${borderRadiusMedium};
+    box-shadow: inset 0 0 0 2px ${colorStrokeFocus2};
+    content: '';
+    cursor: pointer;
+    inset: 0px;
+    outline: none;
+    position: absolute;
+  }
+  :host(:hover:not([disabled])) .control {
+    border-color: ${colorNeutralStrokeAccessibleHover};
+  }
+  :host(:active:not([disabled])) .control {
+    border-color: ${colorNeutralStrokeAccessiblePressed};
+  }
+  :host([aria-checked='true']:not([disabled])) .checked-indicator {
+    opacity: 1;
+  }
+  :host([aria-checked='true']:not([disabled])) .control {
+    border-color: var(--control-border-color);
+  }
+  :host([aria-checked='true']:not([disabled])) .checked-indicator {
+    background-color: var(--checked-indicator-background-color);
+  }
+  :host([aria-checked='true']:hover:not([disabled])) .control {
+    border-color: ${colorCompoundBrandStrokeHover};
+  }
+  :host([aria-checked='true']:hover:not([disabled])) .checked-indicator {
+    background-color: ${colorCompoundBrandStrokeHover};
+  }
+  :host([aria-checked='true']:active:not([disabled])) .control {
+    border-color: ${colorCompoundBrandStrokePressed};
+  }
+  :host([aria-checked='true']:active:not([disabled])) .checked-indicator {
+    background: ${colorCompoundBrandForeground1Pressed};
+  }
```

Comment:
It would be great if we could find a way to reduce the usage of `:not()` as it increases complexity in terms of resolution here. These are fine, but we ideally can find ways to increase specificity for disabled instead of significantly increasing the specificity of every enabled selector.